paymentsheet/src/androidTest/java/com/stripe/android/paymentsheet/GooglePayTest.kt

```diff
@@ -0,0 +1,158 @@
+package com.stripe.android.paymentsheet
+
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.hasTestTag
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import com.google.testing.junit.testparameterinjector.TestParameter
+import com.google.testing.junit.testparameterinjector.TestParameterInjector
+import com.stripe.android.googlepaylauncher.GooglePayRepository
+import com.stripe.android.networktesting.RequestMatchers.host
+import com.stripe.android.networktesting.RequestMatchers.method
+import com.stripe.android.networktesting.RequestMatchers.path
+import com.stripe.android.networktesting.ResponseReplacement
+import com.stripe.android.networktesting.testBodyFromFile
+import com.stripe.android.paymentsheet.ui.GOOGLE_PAY_BUTTON_TEST_TAG
+import com.stripe.android.paymentsheet.ui.PAYMENT_SHEET_FORM_TEST_TAG
+import com.stripe.android.paymentsheet.ui.SAVED_PAYMENT_METHOD_CARD_TEST_TAG
+import com.stripe.android.paymentsheet.ui.SAVED_PAYMENT_OPTION_TAB_LAYOUT_TEST_TAG
+import com.stripe.android.paymentsheet.utils.ProductIntegrationType
+import com.stripe.android.paymentsheet.utils.ProductIntegrationTypeProvider
+import com.stripe.android.paymentsheet.utils.TestRules
+import com.stripe.android.paymentsheet.utils.expectNoResult
+import com.stripe.android.paymentsheet.utils.runProductIntegrationTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(TestParameterInjector::class)
+internal class GooglePayTest {
+    @TestParameter(valuesProvider = ProductIntegrationTypeProvider::class)
+    lateinit var integrationType: ProductIntegrationType
+
+    @get:Rule
+    val testRules: TestRules = TestRules.create()
+
+    private val composeTestRule = testRules.compose
+
+    @Test
+    fun googlePayIsShownWhenFullyEnabled() = runGooglePayTest(
+        isGooglePayReady = true,
+        isGooglePayEnabledInElementsSession = true,
+        hasGooglePayConfig = true,
+    ) {
+        composeTestRule.onGooglePayOption().assertExists()
+    }
+
+    @Test
+    fun googlePayIsNotShownWhenNotReady() = runGooglePayTest(
+        isGooglePayReady = false,
+        isGooglePayEnabledInElementsSession = true,
+        hasGooglePayConfig = true,
+    ) {
+        composeTestRule.onGooglePayOption().assertDoesNotExist()
+    }
+
+    @Test
+    fun googlePayIsNotShownWhenNotEnabledInElementsSession() = runGooglePayTest(
+        isGooglePayReady = true,
+        isGooglePayEnabledInElementsSession = false,
+        hasGooglePayConfig = true,
+    ) {
+        composeTestRule.onGooglePayOption().assertDoesNotExist()
+    }
+
+    @Test
+    fun googlePayIsNotShownWhenGooglePayConfigIsNotProvided() = runGooglePayTest(
+        isGooglePayReady = true,
+        isGooglePayEnabledInElementsSession = true,
+        hasGooglePayConfig = false,
+    ) {
+        composeTestRule.onGooglePayOption().assertDoesNotExist()
+    }
```

Comment:
Added just availability tests for now. Follow-up will have intent confirmation tests.